### PR TITLE
#8 subscription manager

### PR DESCRIPTION
**Pull request by sponnet**
subscription manager

### CODE REVIEW

File: API/socket.js [hunk 1]
```diff
@@ -123,9 +123,48 @@ io.on('connection', (socket) => {
...
+const APIHOST = process.env.APIHOST || '0.0.0.0';
+/**
+ * start the socket server and start listening
+ *
+ * @return     {Promise}  { resolves with { port , host} when listening }
+ */
+function listen() {
+	return new Promise((resolve, reject) => {
+		if (!APISOCKETPORT || !APIHOST) {
+			return reject('no APISOCKETPORT defined in environment');
```
Comment (bkawk) [inline API/socket.js]: I don't think we need to return the reject here

Comment (bkawk) [reply API/socket.js]: Should we also log this?

File: API/socket.js [hunk 2]
```diff
@@ -123,9 +123,48 @@ io.on('connection', (socket) => {
...
+		}
+		server.listen(APISOCKETPORT, APIHOST, (err) => {
+			if (err) {
+				return reject(err);
+			}
+			logs.info('server listening on host ', APIHOST, 'port', APISOCKETPORT);
+			return resolve({
```
Comment (bkawk) [inline API/socket.js]: again I don't think we need to return the resolve

File: API/socket.js [hunk 3]
```diff
@@ -123,9 +123,48 @@ io.on('connection', (socket) => {
...
+/**
+ * stop listening for new connections
+ *
+ * @return     {Promise}  { description_of_the_return_value }
+ */
+function close() {
+	return new Promise((resolve, reject) => {
+		server.close((err) => {
+			if (err) {
+				return reject(err);
```
Comment (bkawk) [inline API/socket.js]: again here

File: API/socket.js [hunk 4]
```diff
@@ -123,9 +123,48 @@ io.on('connection', (socket) => {
...
+			}
+			return resolve();
```
Comment (bkawk) [inline API/socket.js]: and here

**Review by bkawk — CHANGES_REQUESTED** [review]
Looks really good, the only comments were in the starting and stopping of socket.js
1) We don't need to return a resolve or reject
2) We should consider if we need to log any issues for successes with starting and stopping

**Comment by bkawk** [comment]
From the front end im also getting a connection error `localhost:8011/socket.io/?publicKey=0x7d475940895c7fa932db98844da201ee86d0568d&EIO=3&transport=polling&t=L-Pv-0s:1 GET http://localhost:8011/socket.io/?publicKey=0x7d475940895c7fa932db98844da201ee86d0568d&EIO=3&transport=polling&t=L-Pv-0s net::ERR_CONNECTION_REFUSED`

File: API/socket.js (hunk 1 above)
Comment (sponnet) [reply API/socket.js]: done

File: API/socket.js (hunk 2 above)
Comment (sponnet) [reply API/socket.js]: done

File: API/socket.js (hunk 3 above)
Comment (sponnet) [reply API/socket.js]: done

File: API/socket.js (hunk 4 above)
Comment (sponnet) [reply API/socket.js]: ..and done

**Pull request merged**